python/ray/tests/test_basic_2.py

```diff
@@ -558,6 +558,44 @@ def add(self, x):
     assert r1 == r2 == r3
 
 
+def test_actor_max_concurrency(ray_start_regular):
+    """
+    Test that an actor of max_concurrency=N should only run
+    N tasks at most concurrently.
+    """
+    CONCURRENCY = 5
+
+    @ray.remote
+    class ConcurentActor:
+        def __init__(self):
+            self.concurrent_tasks = 0
+            self.lock = threading.Lock()
+            self.max_concurrency = 0
+
+        def call(self):
+            with self.lock:
+                self.concurrent_tasks += 1
+                self.max_concurrency = max(self.max_concurrency,
+                                           self.concurrent_tasks)
+            time.sleep(0.5)
```

Comment:
let's avoid sleeping here please. you can use the `ray.test_utils.SignalActor` to control when to block/unblock explicitly